backend/src/v2/driver/driver.go

```diff
@@ -604,32 +623,52 @@ func extendPodSpecPatch(
 
 		for _, toleration := range tolerations {
 			if toleration != nil {
-				k8sToleration := k8score.Toleration{
-					Key:               toleration.Key,
-					Operator:          k8score.TolerationOperator(toleration.Operator),
-					Value:             toleration.Value,
-					Effect:            k8score.TaintEffect(toleration.Effect),
-					TolerationSeconds: toleration.TolerationSeconds,
+				k8sToleration := &k8score.Toleration{}
+				if toleration.TolerationJson != nil {
+					err := resolveK8sJsonParameter(ctx, opts, dag, pipeline, mlmd,
+						toleration.GetTolerationJson(), inputParams, k8sToleration)
+					if err != nil {
+						return fmt.Errorf("failed to resolve toleration: %w", err)
+					}
+				} else {
+					k8sToleration.Key = toleration.Key
+					k8sToleration.Operator = k8score.TolerationOperator(toleration.Operator)
+					k8sToleration.Value = toleration.Value
+					k8sToleration.Effect = k8score.TaintEffect(toleration.Effect)
+					k8sToleration.TolerationSeconds = toleration.TolerationSeconds
 				}
-
-				k8sTolerations = append(k8sTolerations, k8sToleration)
+				k8sTolerations = append(k8sTolerations, *k8sToleration)
 			}
 		}
-
 		podSpec.Tolerations = k8sTolerations
 	}
 
 	// Get secret mount information
 	for _, secretAsVolume := range kubernetesExecutorConfig.GetSecretAsVolume() {
+		var secretName string
+		if secretAsVolume.SecretNameParameter != nil {
+			resolvedSecretName, err := resolveK8sParameter(ctx, opts, dag, pipeline, mlmd,
+				secretAsVolume.SecretNameParameter, inputParams)
+			if err != nil {
+				return fmt.Errorf("failed to resolve secret name: %w", err)
+			}
+			secretName = resolvedSecretName.GetStringValue()
+		} else {
+			secretName = secretAsVolume.SecretName
```

Comment:
It seems like based on the proto definition that it's possible the user didn't specify `secretAsVolume.SecretNameParameter` or `secretAsVolume.SecretName`. So you may need to either return error if both are unset or skip it.